pkg/reconciler/metric/metric.go

```diff
@@ -60,8 +60,19 @@ func (r *reconciler) Reconcile(ctx context.Context, key string) error {
 		return errors.Wrapf(err, "failed to fetch metric %q", key)
 	}
 
+	metric.Status.InitializeConditions()
+
 	if err := r.collector.CreateOrUpdate(metric); err != nil {
+		if err.Error() == "failed to get endpoints" {
```

Comment:
I'd rather we switched to typed errors (e.g. collector.ErrFailedGetEndpoints) and then check equality.